scripts/download_models.sh

```diff
@@ -11,27 +11,55 @@ SOURCE_DIR="$CURRENT_DIR/.."
 ARIA_OPTIONS="-x 8 -s 8 --continue --file-allocation=none"
 
 # Set directories
-MODELS_DIR="$SOURCE_DIR/models"
-INPUTS_DIR="$SOURCE_DIR/inputs"
+TINY_MODELS_DIR="$SOURCE_DIR/openai_whisper-tiny"
+BASE_MODELS_DIR="$SOURCE_DIR/openai_whisper-base"
+SMALL_MODELS_DIR="$SOURCE_DIR/openai_whisper-small"
 
-# Make sure folders exist
-mkdir -p "$MODELS_DIR"
-mkdir -p "$INPUTS_DIR"
+function SAFE_MODEL_DIRECTORY(){
+    if [ ! -d "${1}" ]; then
+        echo "mkdir ${1} .."
+        mkdir -p "${1}"
+    fi
+}
+
+SAFE_MODEL_DIRECTORY $TINY_MODELS_DIR
+SAFE_MODEL_DIRECTORY $BASE_MODELS_DIR
+SAFE_MODEL_DIRECTORY $SMALL_MODELS_DIR
 
 # Download Whisper auxiliary models
 HF_ARGMAX_URL="https://huggingface.co/argmaxinc/whisperkit-android/resolve/main"
 
-aria2c $ARIA_OPTIONS -d "$INPUTS_DIR" -o converted_vocab.json $HF_ARGMAX_URL/converted_vocab.json
-aria2c $ARIA_OPTIONS -d "$MODELS_DIR" -o melspectrogram.tflite $HF_ARGMAX_URL/melspectrogram.tflite
-aria2c $ARIA_OPTIONS -d "$MODELS_DIR" -o postproc.tflite $HF_ARGMAX_URL/postproc.tflite
-aria2c $ARIA_OPTIONS -d "$MODELS_DIR" -o voice_activity_detection.tflite $HF_ARGMAX_URL/voice_activity_detection.tflite
+if [ ! -f $TINY_MODELS_DIR/converted_vocab.json ]; then
+    aria2c $ARIA_OPTIONS -d "$TINY_MODELS_DIR" -o converted_vocab.json $HF_ARGMAX_URL/converted_vocab.json
+    aria2c $ARIA_OPTIONS -d "$TINY_MODELS_DIR" -o MelSpectrogram.tflite $HF_ARGMAX_URL/melspectrogram.tflite
+    aria2c $ARIA_OPTIONS -d "$TINY_MODELS_DIR" -o postproc.tflite $HF_ARGMAX_URL/postproc.tflite
+    aria2c $ARIA_OPTIONS -d "$TINY_MODELS_DIR" -o voice_activity_detection.tflite $HF_ARGMAX_URL/voice_activity_detection.tflite
+fi
+if [ ! -f $BASE_MODELS_DIR/converted_vocab.json ]; then
+    cp $TINY_MODELS_DIR/* $BASE_MODELS_DIR/.
```

Comment:
Does this copy over all of the contents of TINY_MODELS_DIR to BASE_MODELS_DIR? Wouldn't this include the model files [AudioEncoder.tflite, TextDecoder.tflite], not just converted_vocab.json?

Reply:
That's a correct concern. Most likely if any single file is missing in those model folder, it means it hasn't downloaded any before. And AudioEncoder.tflite and TextDecoder.tflite are downloaded after this. 
Do you still have a concern? Then I can modify it so that it doesn't overwrite OpenAI models.

Reply:
I updated it per the review. Thanks.